### ISSUE TITLE
The protonvpn-cli reconnect service fails on start

### ISSUE THREAD

**Issue by DeskworkTrickster** [NixOS/nixpkgs#182071]
### Describe the bug
The `protonvpn_reconnect.service` fails on start.

### Steps To Reproduce
1. Try to tart the `protonvpn_reconnect.service`

### Expected behavior
`protonvpn_reconnect.service` starts without errors.

### Additional context
Jul 19 07:57:25 nixos-fw135 systemd[1353]: Started Proton VPN Reconnector.
Jul 19 07:57:25 nixos-fw135 python3.10[77587]: Traceback (most recent call last):
Jul 19 07:57:25 nixos-fw135 python3.10[77587]:   File "/nix/store/dlcmn14jm9xj06mj5igb7db862q2hi90-python3.10-protonvpn-nm-lib-3.10.0/lib/python3.10/site-packages/protonvpn_nm_lib/daemon/dbus_daemon_reconnector.py", line 3, in <module>
Jul 19 07:57:25 nixos-fw135 python3.10[77587]:     import dbus
Jul 19 07:57:25 nixos-fw135 python3.10[77587]: ModuleNotFoundError: No module named 'dbus'
Jul 19 07:57:26 nixos-fw135 systemd[1353]: protonvpn_reconnect.service: Main process exited, code=exited, status=1/FAILURE
Jul 19 07:57:26 nixos-fw135 systemd[1353]: protonvpn_reconnect.service: Failed with result 'exit-code'.

### Notify maintainers
@WolfangAukang

**Comment by WolfangAukang** [comment]
Are you picking this from the stable branch? This issue is related to protonvpn-nm-lib missing the dbus dependencies. A backport PR has been set [here](https://github.com/NixOS/nixpkgs/pull/176817), and the PR with the fix is located [here](https://github.com/NixOS/nixpkgs/pull/172939).

**Comment by DeskworkTrickster** [comment]
just wondering, should this be alrdy fixed when running unstable?
i updated this morning but the error is still the same.

**Comment by WolfangAukang** [comment]
Until now I could take a closer look at this issue. Checking the content of the service file (currently from 3.11.0, but the functionality is similar on any version being built):

```
> cat /home/hamadare/.config/systemd/user/protonvpn_reconnect.service

# v3.11.0

[Unit]
Description=Proton VPN Reconnector

[Service]
ExecStart=/nix/store/553d7c4xcwp9j1a1gb9cb1s9ry3x1pi9-python3-3.9.13/bin/python3.9 /nix/store/s81il9wdkl30iji64hvhk8b1mfxm40zm-python3.9-protonvpn-nm-lib-3.11.0/lib/python3.9/site-packages/protonvpn_nm_lib/daemon/dbus_daemon_reconnector.py
```

My guess is that as this is using the Python separately, it is not picking the propagated build inputs from the protonvpn-nm-lib package somehow, because as you can see below, it is there (first one):

```
cat /nix/store/36ycpnh736p0czixqri2gshhg27dvxpk-python3.10-protonvpn-nm-lib-3.10.0/nix-support/propagated-build-inputs
/nix/store/3v4ffcb0jszfhl54ayw4rkfbri3hq3g1-python3.10-dbus-python-1.2.18-dev /nix/store/kmwrl75mfcq6mnv6gffz1dm40f6xf431-python3.10-distro-1.7.0 /nix/store/25zqdhcncrz4p38z6ih4gdc111mch6sk-python3.10-Jinja2-3.1.2 /nix/store/60yaizc6ggb89bh0bvc58ihmbwpr4xzm-python3.10-keyring-23.6.0 /nix/store/47mlqa858ljjai3qpl9iw6yzc6jvkxri-python3.10-proton-client-0.7.1 /nix/store/f37zi0mnmmdx9r6lifwqw3sgbd77g1as-python3.10-pygobject-3.42.1-dev /nix/store/v0x19hrdfpnfsv7z5sljds9j9c3z1ifm-python3.10-pyxdg-0.27 /nix/store/90n0lxbnflljjvkryr423x12czz2iddb-python3.10-systemd-234 /nix/store/43g531pxivh5afgbs9ikdh5bkf5sp835-ncurses-6.3-p20220507-dev /nix/store/n8rp1sdc4303g38vsbrwhcl1gqij0kzz-networkmanager-1.38.2-dev /nix/store/8ld33g4b3v7nqxp0gznd155xc7zp2n6d-systemd-250.4-dev /nix/store/xz432xs7qjpkzl1nfanmqlnslclqx76w-xdg-utils-unstable-2020-10-21 /nix/store/ddhcwc8k9ma093n5vlpa9l617p4q2cax-python3-3.10.5
```

I will check if there are any examples where the service is rewritten to handle this.

**Comment by WolfangAukang** [comment]
So at least I'm able to wrap the `dbus_daemon_reconnector.py` file into a script (which is working) that can be used for the service.

The main issue I'm facing is that this service file is generated when you start a connection with any of the ProtonVPN services (be it CLI or GUI), not during the installation (which would be the preferred course of action). I know on the `protonvpn-nm-lib' code [where this service file is written](https://github.com/ProtonVPN/protonvpn-nm-lib/blob/6a3740fd38cbf21ee98073782b056c653550c568/protonvpn_nm_lib/core/dbus/dbus_reconnect.py#L35), but patching the ExecStart code is not working.

The easiest option in mind would be, on the package definition, write the service file there (which is a very dirty solution in my opinion).

**Comment by WolfangAukang** [comment]
#176817 contains the fixes to the issue

**Comment by DeskworkTrickster** [comment]
ah nice. so since it got merged a few minutes ago: how long does it normally take to arrive in unstable?

**Comment by WolfangAukang** [comment]
You can check it [here](https://nixpk.gs/pr-tracker.html?pr=176817).

Remember to remove the service on `~/.config/systemd/user/`. If you don't remove it, it will still use the old one.

**Comment by WolfangAukang** [comment]
Just in case, to confirm it is working for me (running from the most recent commit on nixpkgs repo):
```
> nix-build -A protonvpn-cli
/nix/store/2jl99xz5jmmlj5aq2w8lx1w2pxl1r3g0-protonvpn-cli-3.12.0
> rm ~/.config/systemd/user/protonvpn_reconnect.service # Remove old service
> ./result/bin/protonvpn-cli c
Setting up Proton VPN.
Connecting to Proton VPN on US-FREE#24 with UDP.

Successfully connected to Proton VPN.
> cat ~/.config/systemd/user/protonvpn_reconnect.service

# v3.11.0

[Unit]
Description=Proton VPN Reconnector

[Service]
ExecStart=/nix/store/xyi3lhkrmsd4syvqclzjn6q2h4plbhvi-python3.10-protonvpn-nm-lib-3.11.0/bin/protonvpn_reconnector.py
> systemctl status --user protonvpn_reconnect.service
● protonvpn_reconnect.service - Proton VPN Reconnector
     Loaded: loaded (/home/hamadare/.config/systemd/user/protonvpn_reconnect.service; static)
     Active: active (running) since Sun 2022-07-31 14:57:16 CST; 24s ago
   Main PID: 167103 (python3.10)
      Tasks: 1 (limit: 38442)
     Memory: 12.0M
        CPU: 129ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/protonvpn_reconnect.service
             └─167103 /nix/store/9wa02q541sxq7372f8zv27rl57aribxj-python3-3.10.5/bin/python3.10 /nix/store/xyi3lh>

Jul 31 14:57:16 holuhraun systemd[2137]: Started Proton VPN Reconnector.
```

This could help in verifying if it is working for you.

**Comment by DeskworkTrickster** [comment]
Thanks a lot. So far it works fine now.
It's strictly reconnecting? And auto connecting to a VPN server at startup is another thing?
Just wanna make sure, since I have a startup script for that already and don't want to double connecting. :)

**Comment by WolfangAukang** [comment]
Unfortunately this is something I might not be able to answer, specially because before this issue I was not aware the application had a service file. But from what I understand, what I just did was replacing the execution path from the ExecStart (which was using a Python environment without the necessary dependencies) with the one from the wrapped script (which does contain Python with the necessary dependencies, in this case, `python-dbus`). And from what I have tested so far, the service starts/stops when connecting/disconnecting. So not much has been changed functionally.

I will close the issue as you have indicated it has been solved.
